rlclientlib/live_model_impl.h

```diff
@@ -104,6 +104,10 @@ namespace reinforcement_learning
     std::unique_ptr<model_management::i_data_transport> _transport{nullptr};
     std::unique_ptr<model_management::i_model> _model{nullptr};
 
+    // These objects need to be handled VERY carefully. _interaction_logger will spawn a thread
+    // which will contain a pointer to the _logger_extensions object. The thread's lifetime is
+    // tied to the _interaction_logger object. If any of these conditions change, the _logger_extentions
```

Comment:
Why not just make it a shared object now so we don't have this scary message?

Reply:
> I am not very familiar with the code this is changing so I am finding reviewing this a bit tricky. The main question I have is, now that the code is done do we understand what the performance improvements are over the previous implementation? Can we measure this to get a picture?

I'll try to get some performance metrics, but it'll be a bit tricky since I'll either have to figure out how to stub out the logger sink or we'll have I/O noise to deal with

Reply:
> Why not just make it a shared object now so we don't have this scary message?

I could, I just found it to be unnecessary in this case. @rajan-chari How would you feel about switching this case to a shared_ptr?

Reply:
The concern for proper cleanup order is prudent.  rlclientlib thread and object lifetimes were carefully designed for proper cleanup.  i.e. thread are signaled to cleanup and join during shutdown before all resources are deallocated.  If we find that this is not the case, it should be fixed.

Reply:
I believe in the next sprint, we are auditing suspect shutdown hygiene issue in rlclientlib.  Might be a good time to confirm that things still work as designed.

Reply:
The general idea is that live_model_impl has ownership of all these resources and should clean them up after proper shutdown of all background threads.

Reply:
unique_ptr signals this contract and design.  I'd like to leave that in place.